packages/date-picker/src/vaadin-date-picker-overlay-content.js

```diff
@@ -855,15 +863,29 @@ class DatePickerOverlayContent extends ControllerMixin(ThemableMixin(DirMixin(Po
     if (!keepMonth) {
       this._focusedMonthDate = dateToFocus.getDate();
     }
-    await this.focusDateElement();
+    await this.focusDateElement(false);
   }
 
-  async focusDateElement() {
+  async focusDateElement(reveal = true) {
```

Comment:
nit: untested `reveal` flag

Reply:
Added two tests, one to cover `.focusDateElement(false)` that happens when navigating days, and another one to cover `.focusDateElement()` call with no flag set. Note, the latter is also implicitly tested here:

https://github.com/vaadin/web-components/blob/71104da9006ce9df6eb310ad5e95a9844bb6fab6/packages/date-picker/test/keyboard-input.test.js#L240-L241